src/catalog/__tests__/catalogEntities.spec.js

```diff
@@ -0,0 +1,108 @@
+import mockConsole from 'jest-mock-console';
+import { apiKinds, apiCredentialKinds } from '../../api/apiConstants';
+import * as catalogEntities from '../catalogEntities';
+
+describe('/renderer/catalogEntities', () => {
+  beforeEach(() => {
+    mockConsole(); // automatically restored after each test
+  });
+
+  describe('generateEntityUrl()', () => {
+    [
+      apiKinds.RHEL_LICENSE,
+      apiKinds.PROXY,
+      apiKinds.PUBLIC_KEY,
+      apiKinds.CLUSTER,
+      ...Object.values(apiCredentialKinds),
+    ].forEach((apiKind) => {
+      const entity = {
+        metadata: {
+          cloudUrl: 'https://test-cloud-url.com',
+          kind: apiKind,
+          name: 'test-entity-name',
+          namespace: 'test-namespace',
+        },
+      };
+
+      it(`returns url for |${apiKind}| entity`, () => {
+        const url = `${entity.metadata.cloudUrl}/projects/${entity.metadata.namespace}`;
+
+        if (Object.values(apiCredentialKinds).includes(entity.metadata.kind)) {
+          expect(catalogEntities.generateEntityUrl(entity)).toEqual(
+            `${url}/creds?name=${entity.metadata.name}&details`
+          );
+        }
+
+        switch (entity.metadata.kind) {
+          case apiKinds.RHEL_LICENSE:
+            expect(catalogEntities.generateEntityUrl(entity)).toEqual(
+              `${url}/rhel?name=${entity.metadata.name}&details`
+            );
+            break;
+          case apiKinds.PROXY:
+            expect(catalogEntities.generateEntityUrl(entity)).toEqual(
+              `${url}/proxies?name=${entity.metadata.name}&details`
+            );
+            break;
+          case apiKinds.PUBLIC_KEY:
+            expect(catalogEntities.generateEntityUrl(entity)).toEqual(
+              `${url}/keypairs?name=${entity.metadata.name}&details`
+            );
+            break;
+          case apiKinds.CLUSTER:
+            expect(catalogEntities.generateEntityUrl(entity)).toEqual(
+              `${url}/clusters/${entity.metadata.name}`
+            );
+            break;
+          default:
+            if (!Object.values(apiCredentialKinds).includes(apiKind)) {
+              expect(catalogEntities.generateEntityUrl(entity)).toEqual(url);
+            }
```

Comment:
Shouldn't it be this?

```suggestion
            if (Object.values(apiCredentialKinds).includes(apiKind)) {
              expect(catalogEntities.generateEntityUrl(entity)).toEqual(url);
            } else {
              throw new Error(`Unexpected apiKind="${apiKind}": Test case may need to be enhanced to support it`);
            }
```

As it is, I don't think we're testing any of the credential kinds...